code/modules/mob/living/carbon/xenomorph/xeno_defines.dm

```diff
@@ -927,13 +927,34 @@
 	hivecore_cooldown = FALSE
 	xeno_message(SPAN_XENOBOLDNOTICE("The weeds have recovered! A new hive core can be built!"),3,hivenumber)
 
-/datum/hive_status/proc/free_respawn(client/C)
+/datum/hive_status/proc/free_respawn(client/C, turf/spawning_turf)
 	stored_larva++
 	if(!hive_location || !hive_location.spawn_burrowed_larva(C.mob))
 		stored_larva--
+		respawn_on_turf(C, spawning_turf)
 	else
 		hive_ui.update_burrowed_larva()
```

Comment:
```suggestion
/datum/hive_status/proc/free_respawn(client/xeno_client, turf/spawning_turf)
	stored_larva++
	if(!hive_location || !hive_location.spawn_burrowed_larva(xeno_client.mob))
		stored_larva--
		return respawn_on_turf(xeno_client, spawning_turf)
	hive_ui.update_burrowed_larva()
	return TRUE
```

Does this proc even need to be changed though? You didn't end up using it for this.